flexget/plugins/sites/hliang.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.plugins.internal.urlrewriting import UrlRewritingError
+from flexget.utils.soup import get_soup
+
+log = logging.getLogger('hliang')
+
+class UrlRewriteHliang(object):
+    """Hliang urlrewriter."""
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.startswith('http://bt.hliang.com/show.php'):
+            return True
+        return False
+
+    # urlrewriter API
+    def url_rewrite(self, task, entry):
+        entry['url'] = self.parse_download_page(entry['url'], task.requests)
+
+    @plugin.internet(log)
+    def parse_download_page(self, url, requests):
+        txheaders = {'User-agent': 'Mozilla/4.0 (compatible; MSIE 5.5; Windows NT)'}
+        try:
+            page = requests.get(url, headers=txheaders)
```

Comment:
1. Just that a plugin code is what it is, doesn't mean it's OK. We try to constantly improve the code quality of stuff, especially with new PRs where it's easier (since the plugin dev is still around).
2. I didn't mean that you shouldn't raise a UrlRewritingError, I meant that if possible, avoiding catching all Exception, since that can swallow up stuff you maybe would like to treat differently. Again, this is more of a global guide line than a rule, and it certainly can be that this fits here.
3. By having as minimal code as possible in each `try`, you make sure the outcome is expected. This is especially true since you are catching `Exception` in the second part, which could be more relevant to the soup part than request part. So if ANY request error that isn't expected we will still raise that error. 

I hope this comes off in the manner which I mean it to, which is just trying to improve the global health of the plugin and not deter you from this in any way.